### PR TITLE
Refactor code to use auto-generated id for user profiles

### PR DESCRIPTION
**Pull request by master-faz**
I think I made all the changes to where current user's firebase profile is called. Their email is saved to the profile instead of being the document ID. To find the document query the users collection for a matching email and set a global doc ref to results. In some places where docId is needed for onCreate the id is passed as an intent.

Issue not create for this

### CODE REVIEW

File: app/src/main/java/study_dev/testbed/studyhopper/models/Profile.java [hunk 1]
```diff
@@ -7,32 +7,45 @@
...
-    String university;
-    Timestamp lastLogin;
-
-    Profile() {
 
-    }
+    String email;
 
-    public Profile(String firstName, String lastName, Date dob, String gender, String university) {
+    public Profile(String firstName, String lastName, Date dob, String gender, String university, String email) {
```
Comment (mkaburis) [inline app/src/main/java/study_dev/testbed/studyhopper/models/Profile.java:13]: One suggestion will be adding a screen name to profiles that the user can change. However it is something more for a future update.

**Review by mkaburis — APPROVED** [review]
Changes appear to be in order. Prepare for merge conflicts once we attempt to push both of our pull requests to master.

File: app/src/main/java/study_dev/testbed/studyhopper/models/Profile.java (hunk 1 above)
Comment (master-faz) [reply app/src/main/java/study_dev/testbed/studyhopper/models/Profile.java:13]: that can be done directly from Authentication. It has a display name property. But yeah I can add it when we do profile pics

**Comment by master-faz** [comment]
I'll wait for yours to merge. It'll be easier to do the changes in this branch.

**Pull request merged**